zabbix-agent.sls

```diff
@@ -0,0 +1,22 @@
+# both 32-bit (x86) AND a 64-bit (AMD64) installer available
+{% set versions = {'4.2':['3'], '4.0':[9], '3.0':[28]} %}
+{% set source_path = 'https://www.zabbix.com/downloads/' %}
+
+zabbix-agent:
+{% for major, subversions in versions.items() %}
+{% for minor in subversions %}
+  '{{major}}.{{minor}}.2400':
+    full_name: 'Zabbix Agent (64-bit)'
+    {% if grains['cpuarch'] == 'AMD64' %}
+    installer: '{{source_path}}{{major}}.{{minor}}/zabbix_agent-{{major}}.{{minor}}-win-amd64-openssl.msi'
+    uninstall_flags: '/x {{source_path}}{{major}}.{{minor}}/zabbix_agent-{{major}}.{{minor}}-win-amd64-openssl.msi /quiet'
+    {% else %}
+    installer: '{{source_path}}{{major}}.{{minor}}/zabbix_agent-{{major}}.{{minor}}-win-i386-openssl.msi'
+    uninstall_flags: '/x {{source_path}}{{major}}.{{minor}}/zabbix_agent-{{major}}.{{minor}}-win-i386-openssl.msi /quiet'
```

Comment:
remove the installer from the `uninstall_flags` and use only the parameter `/qn /norestart`

Reply:
That doesn't actually seem to work for this package.  Having the installer
in the flag was the only way it would properly uninstall without
interaction. Even via cli.  /qn /norestart is fine and probably safer but
the only way it would uninstall for me with no interaction was to include
the installer in the flags.  Even using the installer as the uninstall
method instead of msiexec didn't seem to work properly realizing that all
that would be doing is calling msiexec because of the msi package.  Or
using msiexec with the program uuid still didn't work for me.

Reply:
Was finally able to make this work with the msiexec: True flag for the uninstaller.  I must have been including the /x flag for uninstall_flags on the state when it wasn't needed.